pkg/api/validation/validation.go

```diff
@@ -372,6 +372,11 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *api.ObjectMeta, fldPath *field.P
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceVersion"), newMeta.ResourceVersion, "must be specified for an update"))
 	}
 
+	// Generation should be the same or at most incremented by one.
+	if newMeta.Generation != oldMeta.Generation && newMeta.Generation != oldMeta.Generation+1 {
```

Comment:
> Clients should not be able to set generation.

> Only code in apiserver may change generation.

how is that enforced? seems like it should be done centrally, with a `ShouldIncrementGeneration(old,new)`-type method on the update strategy that `BeforeUpdate` can interrogate.